packages/navigator/src/browser/open-editors-widget/navigator-open-editors-tree-model.ts

```diff
@@ -0,0 +1,112 @@
+};
+
+export namespace OpenEditorNode {
+    export function is(node: object | undefined): node is OpenEditorNode {
+        return FileStatNode.is(node) && 'widget' in node;
+    }
+}
+
+@injectable()
+export class OpenEditorsModel extends FileTreeModel {
+    @inject(ApplicationShell) protected readonly applicationShell: ApplicationShell;
+    @inject(WorkspaceService) protected readonly workspaceService: WorkspaceService;
+    @inject(EditorPreviewManager) protected readonly editorPreviewManager: EditorPreviewManager;
+
+    protected toDisposeOnPreviewWidgetReplaced = new DisposableCollection();
+
+    @postConstruct()
+    protected init(): void {
+        super.init();
+        this.initializeRoot();
+    }
+
+    protected async initializeRoot(): Promise<void> {
+        this.selectionService.onSelectionChanged(selection => {
+            const { widget } = (selection[0] as OpenEditorNode);
+            this.applicationShell.activateWidget(widget.id);
+        });
+        this.toDispose.push(this.applicationShell.onDidChangeCurrentWidget(async ({ newValue }) => {
+            const nodeToSelect = this.tree.getNode(newValue?.id) as SelectableTreeNode;
+            if (nodeToSelect) {
+                this.selectNode(nodeToSelect);
+            }
+        }));
+        this.toDispose.push(this.workspaceService.onWorkspaceChanged(() => this.updateOpenWidgets()));
+        this.toDispose.push(this.workspaceService.onWorkspaceLocationChanged(() => this.updateOpenWidgets()));
+        this.toDispose.push(this.applicationShell.onDidAddWidget(() => this.updateOpenWidgets()));
+        this.toDispose.push(this.applicationShell.onDidRemoveWidget(() => this.updateOpenWidgets()));
+        this.toDispose.push(this.editorPreviewManager.onCreated(previewWidget => {
+            if (previewWidget instanceof EditorPreviewWidget) {
+                this.toDisposeOnPreviewWidgetReplaced.dispose();
+                this.toDisposeOnPreviewWidgetReplaced.push(previewWidget.onPinned(() => this.updateOpenWidgets()));
+                this.toDisposeOnPreviewWidgetReplaced.push(previewWidget.onDidChangeTrackableWidgets(() => this.updateOpenWidgets()));
+            }
+        }));
+        await this.updateOpenWidgets();
+        this.fireChanged();
+    }
+
+    protected updateOpenWidgets = debounce(this.doUpdateOpenWidgets, 250);
+
+    protected async doUpdateOpenWidgets(): Promise<void> {
+        const editorWidgets = this.applicationShell.widgets.filter(widget => widget instanceof EditorWidget);
+        this.root = await this.buildRootFromOpenedWidgets(editorWidgets);
+    }
+
+    protected async buildRootFromOpenedWidgets(openWidgets: Widget[]): Promise<CompositeTreeNode> {
+        const newRoot: CompositeTreeNode = {
+            id: 'open-editors:root',
+            parent: undefined,
+            visible: false,
+            children: []
+        };
+        for (const widget of openWidgets) {
+            if (Navigatable.is(widget)) {
```

Comment:
Elsewhere you've checked whether the widget is saveable; why the difference?

Reply:
This was from an early attempt at categorizing the widgets. I've changed this to only check for EditorWidget and EditorPreview widget since those are really the only widgets I'm looking for and that ends up being more consistent with what I'm doing elsewhere.